crates/languages/src/go.rs

```diff
@@ -397,6 +402,128 @@ impl super::LspAdapter for GoLspAdapter {
     }
 }
 
+pub(crate) struct GoContextProvider;
+
+const GO_TEST_DIRECTORY_TASK_VARIABLE: VariableName =
+    VariableName::Custom(Cow::Borrowed("GO_TEST_DIRECTORY"));
+const GO_SUBTEST_NAME_TASK_VARIABLE: VariableName =
+    VariableName::Custom(Cow::Borrowed("GO_SUBTEST_NAME"));
+
+impl ContextProvider for GoContextProvider {
+    fn build_context(
+        &self,
+        _: Option<&Path>,
+        location: &Location,
+        cx: &mut gpui::AppContext,
+    ) -> Result<TaskVariables> {
+        let local_abs_path = location
+            .buffer
+            .read(cx)
+            .file()
+            .and_then(|file| Some(file.as_local()?.abs_path(cx)));
+
+        let test_folder = local_abs_path
+            .as_deref()
+            .and_then(|local_abs_path| local_abs_path.parent())
+            .and_then(|parent| parent.to_string_lossy().into());
+
+        let snapshot = location.buffer.read(cx).snapshot();
+        let point_range = location.range.to_point(&snapshot);
+        let end = Point::new(point_range.start.row + 1, 0);
+        let line = snapshot.text_for_range(point_range.start..end).peek();
```

Comment:
I think it would make sense for `build_context` function to get access to context built up so far; that way, it could inspect variables set by the captures and you wouldn't have to extract function name from current line, which in my opinion isn't ideal. You really had no better tool at your disposal though.
Tomorrow I'll wire through the changes that will allow us to simplify that function a bit.

Reply:
@osiewicz Unfortunately, I haven't found a better way to get the line. As you said, It would be better if I could just access via`build_context`. I would appreciate your assistance 🙏

Reply:
No worries at all! It's not like there was a better way either.
I've opened a PR in https://github.com/zed-industries/zed/pull/12134; once that lands,
you'll be able to capture the first parameter (that's a string) as a hidden variable, process it (stripping "Run" prefix) and then save it as the `SUBTASK_NAME`.